tracer/test/Datadog.Trace.Tests/Propagators/MultiSpanContextPropagatorTests.cs

```diff
@@ -527,6 +527,61 @@ public void ExtractAndInject_B3SingleHeader_PreserveOriginalTraceId()
             headersForInjection.Verify(h => h.Set("b3", expectedTraceParent), Times.Once());
         }
 
+        [Theory]
+        [InlineData(false, false)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        public void W3C_ParentExtracted_WithSameTraceIdAndParentId(bool extractFirst, bool w3CHeaderFirst)
+        {
+            // headers1 equivalent from system-tests
+            var headers = new Mock<IHeadersCollection>();
+
+            headers.Setup(h => h.GetValues("traceparent"))
+                   .Returns(new[] { "00-00000000000000000000000000000001-000000003ade68b1-01" });
```

Comment:
I would feel a little better if we used a 128-bit trace id here, since they are enabled by default and will expect another header, `x-datadog-tags` with the `tid` tag.

Reply:
Or even better, test both cases, 128-bit trace id and 64-bit trace id.

Reply:
Changed the test back based on the latest RFC comments: https://github.com/DataDog/dd-trace-dotnet/pull/5449/commits/d786769f8fb0a5295215126e232772dca956a244